noir_stdlib/src/sha512.nr

```diff
@@ -77,9 +77,11 @@ fn msg_u8_to_u64(msg: [u8; 128]) -> [u64; 16] {
     let mut msg64: [u64; 16] = [0; 16];
 
     for i in 0..16 {
+        let mut msg_field: Field = 0;
         for j in 0..8 {
-            msg64[15 - i] = (msg64[15 - i] << 8) + msg[128 - 8*(i + 1) + j] as u64;
```

Comment:
This gets optimized away into a multiplication by the compiler anyway right? We can keep the change but I find the bit shift more clear. Did this provide an optimization?

Reply:
Yeah, we perform checked arithmetic on integers by default so we are range checking multiple times in a situation where we know that the result cannot overflow. I'm doing the maths inside a field so we only have a single range check at the end.